pkg/client/services.go

```diff
@@ -77,7 +77,13 @@ func (c *services) Get(name string) (result *api.Service, err error) {
 // Create creates a new service.
 func (c *services) Create(svc *api.Service) (result *api.Service, err error) {
 	result = &api.Service{}
-	err = c.r.Post().Namespace(c.ns).Resource("services").Body(svc).Do().Into(result)
+	// v1beta3 does not allow POST without a namespace.
+	needNamespace := !api.PreV1Beta3(c.r.APIVersion())
```

Comment:
If not here, then we will need such conditions everywhere at higher level code.
```
If (apiVersion = v1beta3 && ns == "") {
  ns = api.NamespaceDefault
}
servicesInterface = client.services(ns)
```
Is that better?